pallets/offchain-worker/src/lib.rs

```diff
@@ -557,6 +569,8 @@ pub mod pallet {
 			<ClaimAccountIndex::<T>>::iter().collect::<Vec<_>>().len() as u32
 		}
 
+// SBP M1: move logic in token server. This allows for easier upgrade (no need to redeploy runtime)
+
```

Comment:
the token server just provide the token for ocw via http service. we implement it this way, because no approach found to transfer the token to ocw. we will remove the token server in the future if token could be, for example,  go through from command line to ocw.

Reply:
I'm suggesting to move more logic in the server

Reply:
>> I'm suggesting to move more logic in the server.
If you mean the token server, other process to provide the token (access key). I don't understand why move the code to it.
As I said, we have the server just not find out how to pass the token to ocw, or maybe substrate not provide such method yet. We will definitely remove it in the future. and make the deployment easier.

Reply:
Relying on an external server gives you more flexibility as changes to 3rd party can be handled there and don't force runtime upgrade.

It is very fine to have everything in the runtime if it makes more sense for your use-case.